### ISSUE TITLE
Issue with loading 4bit ipex-llm Mixtral saved model with generation length

### ISSUE THREAD

**Issue by tsantra** [intel-analytics/ipex-llm#11443]
Optimized the Mixtral model by using ipex_llm.optimize_model() to transform it to low-bit and then save it and then load it. 
Set "max_length": 1024 yet getting a warning that `max_length` (=20) .

**_Code:_**

from ipex_llm.optimize import low_memory_init, load_low_bit
from transformers import AutoModelForCausalLM, AutoTokenizer
from ipex_llm.langchain.llms import TransformersLLM
from langchain import PromptTemplate, LLMChain

template ="""{question}"""
prompt = PromptTemplate(template=template, input_variables=["question"])

with low_memory_init():
            model = AutoModelForCausalLM.from_pretrained(save_path_model, torch_dtype="auto", trust_remote_code=True)
        model = load_low_bit(model, save_path_model)
        tokenizer = AutoTokenizer.from_pretrained(save_path_model)  
       
llm = TransformersLLM(model_id="",model=model,tokenizer=tokenizer, model_kwargs={"temperature": 0, "max_length": 1024, "trust_remote_code": True},)

llm_chain = LLMChain(prompt=prompt, llm=llm)
question = "what is AI?"

output = llm_chain.invoke(question)
print("====output=====")
print(output)

**_Output:_**

python3.11/site-packages/langchain_core/_api/deprecation.py:139: LangChainDeprecationWarning: The class `LLMChain` was deprecated in LangChain 0.1.17 and will be removed in 0.3.0. Use RunnableSequence, e.g., `prompt | llm` instead.
  warn_deprecated(
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:2 for open-end generation.
/home/ceed-user/miniforge3/envs/lc-rag-ipex-2/lib/python3.11/site-packages/transformers/generation/utils.py:1355: UserWarning: **_Using the model-agnostic default `max_length` (=20) to control the generation length. We recommend setting `max_new_tokens` to control the maximum length of the generation._**
  warnings.warn(

====output=====
{'question': 'what is AI?', 'text': 'what is AI?\n\nAI is a branch of computer science that deals with the creation of'}

**Comment by shane-huang** [comment]
what is the version your langchain?

**Comment by shane-huang** [comment]
Seems you're running this example on CPU? maybe you can try follow this example to load low bit model. https://github.com/intel-analytics/ipex-llm/blob/main/python/llm/example/CPU/LangChain/low_bit.py

**Comment by lzivan** [comment]
Hi,

We have reviewed your query and used our sample code to address the problem. 

### Solution:
We utilized our sample code to resolve the issue. Here is the relevant section of our sample code that addresses your issue:
```python
import argparse
from ipex_llm.langchain.llms import TransformersLLM, TransformersPipelineLLM
from langchain import PromptTemplate, LLMChain
from langchain import HuggingFacePipeline


def main(args):
    question = args.question
    model_path = args.model_path
    low_bit_model_path = args.target_path
    template ="""{question}"""

    prompt = PromptTemplate(template=template, input_variables=["question"])

    llm = TransformersLLM.from_model_id(
        model_id=model_path,
        model_kwargs={"temperature": 0, "max_length": 1024, "trust_remote_code": True},
    )
    llm.model.save_low_bit(low_bit_model_path)
    del llm
    low_bit_llm = TransformersLLM.from_model_id_low_bit(
        model_id=low_bit_model_path,
        tokenizer_id=model_path,
        model_kwargs={"temperature": 0, "max_length": 1024, "trust_remote_code": True}
    )
    llm_chain = LLMChain(prompt=prompt, llm=low_bit_llm)

    output = llm_chain.run(question)
    print("====output=====")
    print(output)


if __name__ == '__main__':
    parser = argparse.ArgumentParser(description='TransformersLLM Langchain Chat Example')
    parser.add_argument('-m','--model-path', type=str, required=True,
                        help='the path to transformers model')
    parser.add_argument('-t','--target-path',type=str,required=True,
                        help='the path to save the low bit model')
    parser.add_argument('-q', '--question', type=str, default='What is AI?',
                        help='qustion you want to ask.')
    args = parser.parse_args()

    main(args)
```
And then use this command to run it. 

```bash
python low_bit.py -m <path_to_model> -t <path_to_target> [-q <your question>]
```
**Runtime Arguments Explained**:
- `-m MODEL_PATH`: **Required**, the path to the model
- `-t TARGET_PATH`: **Required**, the path to save the low_bit model
- `-q QUESTION`: the question

### Output:
After implementing the solution, we obtained the following output:
```
Artificial Intelligence (AI) refers to the development of computer systems that can perform tasks that normally require human intelligence, such as visual perception, speech recognition, decision-making, and language translation. AI systems use algorithms and machine learning techniques to analyze data, identify patterns, and make decisions based on that data.

There are different types of AI, including:

1. Narrow AI: Also known as weak AI, this type of AI is designed to perform a specific task, such as voice recognition or image recognition.
2. General AI: Also known as strong AI, this type of AI is designed to perform any intellectual task that a human can do.
3. Superintelligent AI: This is a hypothetical type of AI that is much smarter than the best human minds in virtually all spheres, including scientific creativity, general wisdom, and social skills.

AI is used in various industries, including healthcare, finance, manufacturing, transportation, and education, to improve efficiency, accuracy, and productivity. It can also be used to develop new products and services, enhance customer experiences, and solve complex problems.

However, AI also raises ethical concerns, such as privacy, bias, and job displacement. It is important for organizations and individuals to use AI responsibly and ethically, and to consider the potential impact on society as a whole.
====output=====
What is AI?

Artificial Intelligence (AI) refers to the development of computer systems that can perform tasks that normally require human intelligence, such as visual perception, speech recognition, decision-making, and language translation. AI systems use algorithms and machine learning techniques to analyze data, identify patterns, and make decisions based on that data.

There are different types of AI, including:

1. Narrow AI: Also known as weak AI, this type of AI is designed to perform a specific task, such as voice recognition or image recognition.
2. General AI: Also known as strong AI, this type of AI is designed to perform any intellectual task that a human can do.
3. Superintelligent AI: This is a hypothetical type of AI that is much smarter than the best human minds in virtually all spheres, including scientific creativity, general wisdom, and social skills.

AI is used in various industries, including healthcare, finance, manufacturing, transportation, and education, to improve efficiency, accuracy, and productivity. It can also be used to develop new products and services, enhance customer experiences, and solve complex problems.

However, AI also raises ethical concerns, such as privacy, bias, and job displacement. It is important for organizations and individuals to use AI responsibly and ethically, and to consider the potential impact on society as a whole.
```
